src/Metadata/Util/PropertyInfoToTypeInfoHelper.php

```diff
@@ -153,4 +162,146 @@ private static function convertLegacyTypeToType(LegacyType $legacyType): Type
             $legacyType->getCollectionValueTypes(),
         );
     }
+
+    /**
+     * Converts a {@see Type} to what is should have been in the "symfony/property-info" component.
+     *
+     * @return list<LegacyType>|null
+     */
+    public static function convertTypeToLegacyTypes(?Type $type): ?array
```

Comment:
we need to mark this as `@internal`

Reply:
The whole class is internal, isn't it enough?